Marlin/stepper.cpp

```diff
@@ -88,7 +88,10 @@ static volatile char endstop_hit_bits = 0; // use X_MIN, Y_MIN, Z_MIN and Z_MIN_
 #endif
 
 #if PIN_EXISTS(MOTOR_CURRENT_PWM_XY)
-  int motor_current_setting[3] = DEFAULT_PWM_MOTOR_CURRENT;
+  #ifndef PWM_MOTOR_CURRENT
+    #define PWM_MOTOR_CURRENT DEFAULT_PWM_MOTOR_CURRENT
+  #endif
+  const int motor_current_setting[3] = PWM_MOTOR_CURRENT;
```

Comment:
@thinkyhead this line is now in [stepper.h#L137](https://github.com/MarlinFirmware/Marlin/blob/RCBugFix/Marlin/stepper.h#L137). This would fix a compilation error :laughing: ; the error being:
```c
In file included from ./Configuration.h:1235:0,
                 from ./Marlin.h:44,
                 from ./stepper.cpp:47:
./stepper.h: In static member function ‘static void Stepper::digipot_init()’:
./Configuration_adv.h:346:44: error: invalid use of member ‘Stepper::motor_current_setting’ in static member function
 #define PWM_MOTOR_CURRENT {1300, 1630, 1250} // Values in milliamps
                                            ^
./stepper.h:137:44: note: in expansion of macro ‘PWM_MOTOR_CURRENT’
       const int motor_current_setting[3] = PWM_MOTOR_CURRENT;
                                            ^
./stepper.cpp:1009:26: error: from this location
       digipot_current(0, motor_current_setting[0]);
                          ^
In file included from ./Configuration.h:1235:0,
                 from ./Marlin.h:44,
                 from ./stepper.cpp:47:
./Configuration_adv.h:346:44: error: invalid use of member ‘Stepper::motor_current_setting’ in static member function
 #define PWM_MOTOR_CURRENT {1300, 1630, 1250} // Values in milliamps
                                            ^
./stepper.h:137:44: note: in expansion of macro ‘PWM_MOTOR_CURRENT’
       const int motor_current_setting[3] = PWM_MOTOR_CURRENT;
                                            ^
./stepper.cpp:1013:26: error: from this location
       digipot_current(1, motor_current_setting[1]);
                          ^
In file included from ./Configuration.h:1235:0,
                 from ./Marlin.h:44,
                 from ./stepper.cpp:47:
./Configuration_adv.h:346:44: error: invalid use of member ‘Stepper::motor_current_setting’ in static member function
 #define PWM_MOTOR_CURRENT {1300, 1630, 1250} // Values in milliamps
                                            ^
./stepper.h:137:44: note: in expansion of macro ‘PWM_MOTOR_CURRENT’
       const int motor_current_setting[3] = PWM_MOTOR_CURRENT;
                                            ^
./stepper.cpp:1017:26: error: from this location
       digipot_current(2, motor_current_setting[2]);
```
Static member functions can only access static data members.

Here's how Wikipedia explains constant expressions:

> C++ has always had the concept of constant expressions. These are expressions such as 3+4 that will always yield the same results, at compile time and at run time. Constant expressions are optimization opportunities for compilers, and compilers frequently execute them at compile time and hardcode the results in the program. Also, in several places, the C++ specification requires using constant expressions. Defining an array requires a constant expression, and enumerator values must be constant expressions.